c7n/resources/vpc.py

```diff
@@ -134,10 +134,11 @@ def process(self, resources, event=None):
                 for fl in flogs:
                     dest_type_match = (destination_type is None) or op(
                         fl['LogDestinationType'], destination_type)
-                    if 'LogDestination' not in fl:
-                        fl['LogDestination'] = ''
-                    dest_match = (destination is None) or op(
-                        fl['LogDestination'], destination)
+                    if fl.get('LogDestinationType') == "s3":
```

Comment:
Looks like this change will prevent the filter from being able to test cloudwatch destinations. I'm wondering if the original code was close, but needed `fl['LogDestination']` --> `fl.get('LogDestination')` to avoid the `KeyError` you were seeing.